test_crates/type_alias_over_more_generic_underlying/new/src/lib.rs

```diff
@@ -0,0 +1,78 @@
+//! We're going to replace the original type (named like `OriginalType`) with two types:
+//! - `pub struct NewType<A, B = ()>`
+//! - `pub type OriginalType<A> = NewType<A>`
+//!
+//! This is not semver-major, since the type alias has:
+//! - the same name
+//! - the same fields/variants
+//! - the same constructibility and pattern-matching behavior.
+
+pub struct NewStructOnlyPrivateFields<A> {
+    _marker: std::marker::PhantomData<A>,
+}
+
+pub type OriginalStructOnlyPrivateFields = NewStructOnlyPrivateFields<()>;
+
+// ---
+
+pub struct NewStructGeneric<A, B = ()> {
+    _marker: std::marker::PhantomData<(A, B)>,
+}
+
+pub type OriginalStructGeneric<A> = NewStructGeneric<A>;
+
+// ---
+
+pub struct NewStructMixOfFields<A> {
+    _marker: std::marker::PhantomData<A>,
+    pub x: i64,
+}
+
+pub type OriginalStructMixOfFields = NewStructMixOfFields<()>;
+
+// ---
+
+pub struct NewStructConstructible<T> {
+    pub x: T,
+}
+
+pub type OriginalStructConstructible = NewStructConstructible<i64>;
```

Comment:
I think this is the case you need to look into of whether it is semver compatible.  I have a vague recollection of someone warning about replacing a type with `type` and I think it was with `{}` construction syntax.

Reply:
This would be another great use of witness crates (#223), except to demonstrate that some syntax _does not_ break, instead of demonstrating that it breaks.

Reply:
I either found the issue you had in mind or something related: #338

TL;DR: `pub type` replacement of a constructible unit or tuple struct is breaking, since the unit/tuple constructor is not usable on the type alias.

Reply:
[Trait bounds on type aliases aren't checked.](https://github.com/rust-lang/rust/issues/21903)  This is linted (sometimes an error, sometimes a warning), but the lint has false positives.

Additionally -- and as far as I know this is not documented anywhere -- [while lifetime constraints aren't enforced per se, they do have semantic impact.](https://github.com/rust-lang/rust/issues/100270#issuecomment-1293331893)  Therefore, not having the same bounds (or lack of bounds) on the `type` as are on the original struct is a breaking change.  But if it doesn't cause local breakage, people who get the lint warning are likely to apply the suggestion and remove the semantically relevant bounds.

I suspect most people don't know the `dyn Trait` lifetime default mechanism in the first place, and [RFC 2093](https://rust-lang.github.io/rfcs/2093-infer-outlives.html#impact-on-semver) made the potential for breakage a lot more nuanced to boot, so that may be an area to look into independently of `type` aliases as well.